rlclientlib/ranking_event.cc

```diff
@@ -125,4 +125,37 @@ namespace reinforcement_learning {
   const std::string& outcome_event::get_outcome() const { return _outcome; }
   float outcome_event::get_numeric_outcome() const { return _float_outcome; }
   bool outcome_event::get_action_taken() const { return _action_taken; }
+
+  generic_event::generic_event(const char* _id, const timestamp& ts, payload_type type, flatbuffers::DetachedBuffer&& payload, float pass_prob)
+    : _id(_id)
+    , _client_time_gmt(ts)
+    , _payload_type(type)
+    , _payload(std::move(payload))
+    , _pass_prob(pass_prob) {}
+
+  bool generic_event::try_drop(float pass_prob, int drop_pass) {
+    _pass_prob *= pass_prob;
```

Comment:
This code is repeated all over the place, can we unify it?

Reply:
only 2 times - legacy base event class (which should be removed soon) + new generic_event class. We probably can unify it somehow, but I feel it is safer to have paralle v2 implementation on this level. It gives us:
- less risk to break existing v1 implementation
- more flexibility to do things properly in v2 implementation without having dependency on legacy artifacts.

Reply:
Fair point on risk, makes sense.
I don't buy this is a temporary thing. V1 will creeps around for quite a while after v2 goes live, unfortunately.